ed25519/src/lib.rs

```diff
@@ -0,0 +1,34 @@
+extern crate ed25519_dalek;
+extern crate ewasm_api;
+extern crate sha2;
+
+mod verify;
+
+#[cfg(not(test))]
+#[no_mangle]
+pub extern "C" fn main() {
+    let length = ewasm_api::calldata_size();
+
+    ewasm_api::consume_gas(2000);
+
+    if length < 128 {
+        ewasm_api::revert();
+    }
+
+    // FIXME: use ewasm-rust-api 0.6.x to have slices
+    let input = ewasm_api::unsafe_calldata_copy(0, length);
+    let mut tmp = [0u8; 128];
+    tmp.copy_from_slice(&input);
+    match verify::verify(&tmp) {
+        Ok(true) => {
+            ewasm_api::finish_data(&[0x00u8; 4]);
+        }
+        Ok(false) => {
+            ewasm_api::finish_data(&[0xffu8; 4]);
```

Comment:
How does the caller know whether the ed25519 signature was valid if the contract finishes with the same data?

Reply:
One is `0x00` the other is `0xff`.

Reply:
misread it, nvm